raiden/network/transport/matrix/client.py

```diff
@@ -695,15 +697,18 @@ def _handle_responses(
 
                 room = self.rooms[room_id]
                 room.prev_batch = sync_room["timeline"]["prev_batch"]
+                room_members_count = len(room._members)
 
                 for event in sync_room["state"]["events"]:
                     event["room_id"] = room_id
                     room._process_state_event(event)
-
                 for event in sync_room["timeline"]["events"]:
                     event["room_id"] = room_id
                     room._put_event(event)
 
+                # number of members changed. Verify validity of room
+                if room_members_count != len(room._members):
+                    self._handle_member_join_callback(room.room_id)
                 all_messages.append((room, sync_room["timeline"]["events"]))
```

Comment:
The one remaining problem I see with this is that we still will process the messages for a room that the callback will leave.

Reply:
should be already filtered [here](https://github.com/raiden-network/raiden/blob/7bffe949ccc6c7870b6de877b9af4a4cb7236135/raiden/network/transport/matrix/transport.py#L1019-L1031)